rebac-admin-backend/v1/core.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2024 Canonical Ltd.
+
+// Package v1 provides HTTP handlers that implement the ReBAC Admin OpenAPI spec.
+// This package delegates authorization and data manipulation to user-defined
+// "backend"s that implement the designated abstractions.
+package v1
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/canonical/identity-platform-admin-ui/rebac-admin-backend/v1/resources"
+)
+
+// ReBACAdminBackendParams contains references to user-defined implementation
+// of required abstractions, called "backend"s.
+type ReBACAdminBackendParams struct {
+	GroupsService       GroupsServiceBackend
+	GroupsAuthorization GroupsAuthorizationBackend
+}
+
+// ReBACAdminBackend represents the ReBAC admin backend as a whole package.
+type ReBACAdminBackend struct {
+	params  ReBACAdminBackendParams
+	service *service
+}
```

Comment:
No, it's not auto-generated, and it's not because of the embedded struct. That was out of habit, mostly because you can't go wrong with pointers (e.g., there'll be no unexpected struct copies). But in this case we can use a non-pointer field. I'll update the code.